src/control/server/superblock.go

```diff
@@ -0,0 +1,178 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package server
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"gopkg.in/yaml.v2"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/server/ioserver"
+)
+
+const (
+	defaultStoragePath = "/mnt/daos"
+	defaultGroupName   = "daos_io_server"
+	superblockVersion  = 0
+)
+
+// Superblock is the per-Instance superblock
+type Superblock struct {
+	Version     uint8
+	UUID        string
+	System      string
+	Rank        *ioserver.Rank
+	ValidRank   bool
+	MS          bool
+	CreateMS    bool
+	BootstrapMS bool
+}
+
+// TODO: Marshal/Unmarshal using a binary representation?
+
+// Marshal transforms the Superblock into a storable representation
+func (sb *Superblock) Marshal() ([]byte, error) {
+	data, err := yaml.Marshal(sb)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to marshal %+v", sb)
+	}
+	return data, nil
+}
+
+// Unmarshal reconstitutes a Superblock from a Marshaled representation
+func (sb *Superblock) Unmarshal(raw []byte) error {
+	return yaml.Unmarshal(raw, sb)
+}
+
+func (srv *IOServerInstance) superblockPath() string {
+	storagePath := srv.runner.Config.Storage.SCM.MountPoint
+	if storagePath == "" {
+		storagePath = defaultStoragePath
+	}
+	return filepath.Join(srv.fsRoot, storagePath, "superblock")
+}
+
+// NeedsSuperblock indicates whether or not the instance appears
+// to need a superblock to be created in order to start.
+func (srv *IOServerInstance) NeedsSuperblock() (bool, error) {
```

Comment:
why not just call this HasSuperblock? would make more sense

Reply:
If you look at how it's used throughout the code, this name flows better. With HasSuperblock, all of the logic would be negating the result.